File: Easy/Rock Paper Scissors Lizard Spock/Rock Paper Scissors Lizard Spock.cs
using System;
using System.Linq;
using System.Collections.Generic;

class Solution
{
    static void Main(string[] args)
    {
        int N = int.Parse(Console.ReadLine());
        Dictionary<int, char> signs = new Dictionary<int, char>();
        Dictionary<int, string> won = new Dictionary<int, string> ();

        for (int i = 0; i < N; i++)
        {
            string[] inputs = Console.ReadLine().Split(' ');
            signs.Add(int.Parse(inputs[0]),Convert.ToChar(inputs[1]));
        }
        
        while (signs.Count()>1)
        {
            for (int i = 0; i < N/2; i++)
            {   
                int[] result = battleresult(signs.ElementAt(i).Key,
                                        signs.ElementAt(i).Value,
                                        signs.ElementAt(i+1).Key,
                                        signs.ElementAt(i+1).Value);
                
                if (won.ContainsKey(result[1]))
                    won[result[1]] += " " + Convert.ToString(result[0]);
                else
                    won.Add(result[1], Convert.ToString(result[0]));

                signs.Remove(result[0]);
            }
        N/=2;
        }
 
        Console.WriteLine(signs.ElementAt(0).Key);
        Console.WriteLine(won[signs.ElementAt(0).Key]);
    }

static int[] battleresult(int p1, char p1m, int p2, char p2m)
{
    if (p2m ==p1m)
    {
        if (p1<p2)
            return new int[] {p2,p1};
        else
            return new int[] {p1,p2};
    }

    if ((p1m == 'R' && (p2m == 'L' || p2m == 'C')) ||
        (p1m == 'P' && (p2m == 'R' || p2m == 'S')) ||
        (p1m == 'C' && (p2m == 'L' || p2m == 'P')) ||
        (p1m == 'L' && (p2m == 'S' || p2m == 'P')) ||
        (p1m == 'S' && (p2m == 'C' || p2m == 'R')))
        return new int[] {p2,p1};
    else 
        return new int[] {p1,p2};
}                
}

File: Easy/Rock Paper Scissors Lizard Spock/Rock Paper Scissors Lizard Spock.py
import sys
import math

def battleresult(p1, p1m, p2, p2m):
    if p2m ==p1m:
        if p1<p2:
            return (p1,p2)
        else:
            return (p2,p1)

    if ((p1m == 'R' and (p2m == 'L' or p2m == 'C')) or
        (p1m == 'P' and (p2m == 'R' or p2m == 'S')) or
        (p1m == 'C' and (p2m == 'L' or p2m == 'P')) or
        (p1m == 'L' and (p2m == 'S' or p2m == 'P')) or
        (p1m == 'S' and (p2m == 'C' or p2m == 'R'))):
        return (p1,p2)
    else:
        return (p2,p1)             

n = int(input())
signs = {}
won = {}

for i in range(n):
    inputs = input().split()
    signs[int(inputs[0])] = inputs[1]

while len(signs)>1:
    for i in range(0, int(n/2)):  
        win, loose = battleresult(list(signs.items())[i][0],
                                  list(signs.items())[i][1],
                                  list(signs.items())[i+1][0],
                                  list(signs.items())[i+1][1])
                
        if win in won.keys():
            won[win] += " "+str(loose)
        else:
            won[win] = str(loose)

        del signs[loose]

    n/=2
 
print(list(signs.keys())[0])
print(won[list(signs.keys())[0]])